lib/src/neu/verifications/contract-mismatcher.ts

```diff
@@ -423,3 +623,60 @@ export const pathMatchesVariablePath = (
 export class Mismatch {
   constructor(readonly message: string) {}
 }
+
+enum MismatchKind {
+  REQUIRED_HEADER_MISSING = "required_header_missing",
+  UNDEFINED_HEADER = "undefined_header",
+  HEADER_TYPE_MISMATCH = "header_type_mismatch",
+  UNDEFINED_BODY = "undefined_body",
+  BODY_TYPE_MISMATCH = "body_type_mismatch",
+  PATH_PARAM_TYPE_MISMATCH = "path_param_type_mismatch",
+  REQUIRED_QUERY_PARAM_MISSING = "required_query_param_missing",
+  UNDEFINED_QUERY_PARAM = "undefined_query_param",
+  QUERY_PARAM_TYPE_MISMATCH = "query_param_type_mismatch"
+}
+
+class RequiredHeaderMissing {
+  readonly kind = MismatchKind.REQUIRED_HEADER_MISSING;
+  constructor(readonly header: string) {}
+}
+
+class UndefinedHeader {
+  readonly kind = MismatchKind.UNDEFINED_HEADER;
+  constructor(readonly header: string) {}
+}
+
+class HeaderTypeMismatch {
+  readonly kind = MismatchKind.HEADER_TYPE_MISMATCH;
+  constructor(readonly header: string, readonly mismatches: string[]) {}
+}
+
+class UndefinedBody {
+  readonly kind = MismatchKind.UNDEFINED_BODY;
+  constructor() {}
+}
+
+class BodyTypeMismatch {
+  readonly kind = MismatchKind.BODY_TYPE_MISMATCH;
+  constructor(readonly data: string, readonly mismatches: string[]) {}
+}
+
+class PathParamTypeMismatch {
+  readonly kind = MismatchKind.PATH_PARAM_TYPE_MISMATCH;
+  constructor(readonly pathParam: string, readonly mismatches: string[]) {}
+}
+
+class RequiredQueryParamMissing {
+  readonly kind = MismatchKind.REQUIRED_QUERY_PARAM_MISSING;
+  constructor(readonly queryParam: string) {}
+}
+
+class UndefinedQueryParam {
+  readonly kind = MismatchKind.UNDEFINED_QUERY_PARAM;
+  constructor(readonly queryParam: string) {}
+}
+
+class QueryParamTypeMismatch {
```

Comment:
Should these classes be moved to `violations.ts` as well? They seem to be related